Source/Core/Core/AchievementManager.h

```diff
@@ -19,6 +19,8 @@
 #include "Common/WorkQueueThread.h"
 
 using AchievementId = u32;
+constexpr size_t RP_SIZE = 65536;
+using RichPresence = std::array<char, RP_SIZE>;
```

Comment:
I'm skimming the documentation and I'm not really finding any description about the exact format of this parameter. My guess would be that it's just a nullterminated string based on [this function](https://github.com/RetroAchievements/rcheevos/wiki/rc_runtime_get_richpresence), but you seem to be using a fixed-size array, so what's going on here?

Reply:
While it is null-terminated, we have a known maximum length and this can serve as a buffer to that length without needing to be reallocated for size.

Reply:
It would be nice if that maximum length was documented anywhere (or provided in the rcheevos headers), but alright, fair enough.

Reply:
I'm certain it is but I'm having trouble finding it so I'm asking around. We probably should make it more obvious, yes.

Reply:
How is this known in the first place? Looking through the code seems to indicate it has no limit at all, it will go as much as the client is willing to go (well, I guess limited by sizeof(unsigned int) anyways...)

Reply:
Someone on the server linked it: https://docs.retroachievements.org/Rich-Presence/#limits

Reply:
The person also clarified this is a server side enforcement. rcheevos seems to make no such enforcement (and so in theory the server can increase the limit (or an alternative server can be used) and rcheevos would accept that).

Reply:
Wait, aren't we mixing up parameters here? The site claims 65535 characters for the *script*, but this parameter is used for the *displayed text*, no?

On that note, is there a way at all to query the length of the returned string so the caller has the chance to allocate a large enough buffer? Because that would make all of this discussion moot in the first place; we could just ask for the length instead of assuming an arbitrary limit. Usually with such APIs you can call them with a `NULL` buffer to get the length of the required allocation, but that doesn't seem to be the case here, and I'm not seeing any other function related to that either.

Reply:
Like ultimately this is kinda whatever, in any case this is definitely *large enough* for what I expect a little blurb describing the current game state to be, but it still strikes me as a bit... haphazard, y'know?

(Though 64KB is a bit much to have on the stack for a single object, now that I think about it.)

Reply:
Yes, for displayed text it "should" be 255 (although rcheevos again makes no such restriction, that's supposed to be some server side restriction).

There isn't way to query the amount the amount of space needed. In BizHawk, I went the approach of 1024 bytes for a fixed buffer as I figured nothing would go past that and 1KiB is generous enough.

Reply:
Gah, you're right, that's handling the output and not the input, my mistake.
That said, this API requires you to _pass in_ your buffer size, it doesn't return it.

Reply:
It returns the amount of bytes actually written (minus null terminator), which will equal the buffer size - 1 at most. It mimics the C stdlib [snprintf](https://en.cppreference.com/w/c/io/fprintf) function in that way, although in that C function it allows you to put in 0 for the buffer size (and buffer may then be a null pointer), which case it will not write anything but will still return the required size - 1.

Reply:
Specifically, `snprintf` always returns the total amount of characters that would have been written, regardless of the buffer size, which implicitly leads to the 0/NULL case for querying the required buffer size.

Reply:
So is there a further requested change to be made here?